
# HG changeset patch
# User shade
# Date 1582661405 -3600
# Node ID cf0f69f962d94217b1a6c60a2f0ca343500771f7
# Parent  9c948e9fa41e5b5fd169840a4a88e3ef3de67f4c
Fix LRB use in LIRGenerator::do_UnsafeGetAndSetObject
Reviewed-by: rkennke

diff -r 9c948e9fa41e -r cf0f69f962d9 openjdk/hotspot/src/cpu/x86/vm/c1_LIRGenerator_x86.cpp
--- openjdk/hotspot/src/cpu/x86/vm/c1_LIRGenerator_x86.cpp	Fri Dec 06 16:21:26 2019 -0500
+++ openjdk/hotspot/src/cpu/x86/vm/c1_LIRGenerator_x86.cpp	Tue Feb 25 21:10:05 2020 +0100
@@ -1503,10 +1503,8 @@
 
 #if INCLUDE_ALL_GCS
     if (UseShenandoahGC && is_obj) {
-      dst = ShenandoahBarrierSet::barrier_set()->bsc1()->load_reference_barrier(this, dst, NULL, true);
-      LIR_Opr tmp = new_register(type);
-      __ move(dst, tmp);
-      dst = tmp;
+      LIR_Opr tmp = ShenandoahBarrierSet::barrier_set()->bsc1()->load_reference_barrier(this, dst, NULL, true);
+      __ move(tmp, dst);
     }
 #endif
 

